src/python/bcc/__init__.py

```diff
@@ -642,7 +642,10 @@ def get_syscall_prefix(self):
     # return "sys_clone" or "__x64_sys_clone".
     def get_syscall_fnname(self, name):
         name = _assert_is_bytes(name)
-        return self.get_syscall_prefix() + name
+        fullname = self.get_syscall_prefix() + name
+        if self.ksymname(fullname) != -1:
+            return fullname
+        return None
```

Comment:
can we use ksymname() to test return value instead of modifying the function here?

Reply:
OK. I'll move this bit of logic to opensnoop.

BTW, get_syscall_prefix, get_syscall_fnname and fix_syscall_fnname seems a bit weird to me, as they would return a symbol that does not exist at all if all trial had failed. As stated in the comment of get_syscall_prefix, "If no valid value found, will return the first possible value which would probably lead to error in later API calls." It feels returning a None or error is better than returning a non-existing one that might get users confused later.